src/bun.js/test/jest.zig

```diff
@@ -3592,6 +3592,147 @@ pub const Expect = struct {
+            globalThis.throw("toIncludeRepeated() requires the second argument to be a number", .{});
+            return .zero;
+        };
+
+        const expect_string = Expect.capturedValueGetCached(thisValue) orelse {
+            globalThis.throw("Internal consistency error: the expect(value) was garbage collected but it should not have been!", .{});
+            return .zero;
+        };
+
+        if (!expect_string.isString()) {
+            globalThis.throw("toIncludeRepeated() requires the expect(value) to be a string", .{});
+            return .zero;
+        }
+
+        const not = this.op.contains(.not);
+        var pass = std.mem.containsAtLeast(u8, expect_string.toString(globalThis).toSlice(globalThis, default_allocator).slice(), countAsNum, substring.toString(globalThis).toSlice(globalThis, default_allocator).slice());
+
+        if (not) pass = !pass;
+        if (pass) return thisValue;
+
+        var formatter = JSC.ZigConsoleClient.Formatter{ .globalThis = globalThis, .quote_strings = true };
+        const expect_string_fmt = expect_string.toFmt(globalThis, &formatter);
+        const substring_fmt = substring.toFmt(globalThis, &formatter);
+        const times_fmt = count.toFmt(globalThis, &formatter);
+
+        if (not) {
+            const expected_line = "Expected to not include: <green>{any}<r> <green>{any}<r> times\n";
+            const received_line = "Received: <red>{any}<r>\n";
+            const fmt = comptime getSignature("toIncludeRepeated", "<green>expected<r>", true) ++ "\n\n" ++ expected_line ++ received_line;
+            globalThis.throwPretty(fmt, .{ substring_fmt, times_fmt, expect_string_fmt });
+            return .zero;
+        }
+
+        const expected_line = "Expected to include: <green>{any}<r> <green>{any}<r> times \n";
+        const received_line = "Received: <red>{any}<r>\n";
+        const fmt = comptime getSignature("toIncludeRepeated", "<green>expected<r>", false) ++ "\n\n" ++ expected_line ++ received_line;
+        globalThis.throwPretty(fmt, .{ substring_fmt, times_fmt, expect_string_fmt });
+        return .zero;
+    }
+
+    pub fn toSatisfy(this: *Expect, globalThis: *JSGlobalObject, callFrame: *CallFrame) callconv(.C) JSValue {
+        defer this.postMatch(globalThis);
+
+        const thisValue = callFrame.this();
+        const arguments_ = callFrame.arguments(1);
+        const arguments = arguments_.ptr[0..arguments_.len];
+
+        if (arguments.len < 1) {
+            globalThis.throwInvalidArguments("toSatisfy() takes 1 argument", .{});
+            return .zero;
+        }
+
+        if (this.scope.tests.items.len <= this.test_id) {
+            globalThis.throw("toSatisfy() must be called in a test", .{});
+            return .zero;
+        }
+
+        active_test_expectation_counter.actual += 1;
+
+        const right = arguments[0];
```

Comment:
"right" and "left" aren't really descriptive. the expect value can be "value" and the argument can be "predicate"